src/Interpreters/ClientInfo.cpp

```diff
@@ -34,6 +34,12 @@ void ClientInfo::write(WriteBuffer & out, const UInt64 server_protocol_revision)
     writeBinary(initial_query_id, out);
     writeBinary(initial_address.toString(), out);
 
+    if (server_protocol_revision >= DBMS_MIN_PROTOCOL_VERSION_WITH_INITIAL_QUERY_START_TIME)
+    {
+        writeBinary(initial_query_start_time, out);
```

Comment:
Let's not serialize/deserialize both time and time with microseconds as this is redundant. We can serialize just the initial_query_start_time_microseconds and reconstruct the other.

Reply:
Done